### PR TITLE
Adding subscriptions module and adding pool_id to license

### PR DESCRIPTION
**Pull request by john-westcott-iv**
##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - Collection

##### AWX VERSION

```
awx: 21.2.1.dev95+g0a6356a9fa
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx_collection/plugins/modules/subscriptions.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'], 'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: get_subscriptions
```
Comment (sean-m-sullivan) [inline awx_collection/plugins/modules/subscriptions.py]: should be 'subscriptions'

File: awx_collection/plugins/modules/subscriptions.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+RETURN = '''
+subscriptions:
+    description: dictionary containing information about the subscriptions
+    returned: If login succeeded
+    type: dict
+'''
+
+EXAMPLES = '''
+- name: Get subscriptions
+  get_subscriptions:
```
Comment (sean-m-sullivan) [inline awx_collection/plugins/modules/subscriptions.py]: should be 'subscriptions'

File: awx_collection/plugins/modules/subscriptions.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    )
+
+    json_output = {'changed': False}
+
+    # Check if Tower is already licensed
+    post_data = {
+        'subscriptions_password': module.params.get('password'),
+        'subscriptions_username': module.params.get('username'),
+    }
+    json_output['subscriptions'] = module.post_endpoint('config/subscriptions', data=post_data)['json']
```
Comment (sean-m-sullivan) [inline awx_collection/plugins/modules/subscriptions.py]: We should probably filter here, this should work
```
    subscription_data = module.post_endpoint('config/subscriptions', data=post_data)['json']

    subscription_aap_list = []
    for subscription in subscription_data:
        if 'Red Hat Ansible Automation Platform' in subscription['subscription_name']:
            subscription_aap_list.append(subscription)
    json_output['subscriptions'] = subscription_aap_list
    module.exit_json(**json_output)
```

**Comment by sean-m-sullivan** [comment]
I've gone through and tested this on a fresh AAP install, and existing, not previous versions, I think this a great addition. Thinking the filter would be good, but think we may need some input on that, with what subscriptions are named. :+1:

File: awx_collection/plugins/modules/subscriptions.py (hunk 3 above)
Comment (john-westcott-iv) [reply awx_collection/plugins/modules/subscriptions.py]: I added a way to fully filter this but didn't default anything. LMK what you think about that. Also added an example of filtering.

File: awx_collection/plugins/modules/subscriptions.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'], 'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: subscriptions
+author: "John Westcott IV (@john-westcott-iv)"
+short_description: Get subscription list
+description:
+    - Get subscriptions available to  Automation Platform Controller. See
```
Comment (fosterseth) [inline awx_collection/plugins/modules/subscriptions.py:20]: extra space

**Review by fosterseth — APPROVED** [review]

**Review by sean-m-sullivan — APPROVED** [review]
LGTM works good for me.

**Pull request merged**